### PR TITLE
eFAST implementation

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**

### CODE REVIEW

**Comment by Vaibhavdixit02** [comment]
https://www.researchgate.net/publication/243587760_A_Quantitative_Model-Independent_Method_for_Global_Sensitivity_Analysis_of_Model_Output

https://github.com/cran/sensitivity/blob/master/R/fast99.R

https://github.com/SALib/SALib/blob/master/src/SALib/analyze/fast.py

File: src/eFAST.jl [hunk 1]
```diff
@@ -0,0 +1,65 @@
+Base.@kwdef struct eFAST
+    num_harmonics::Int=4
```
Comment (ChrisRackauckas) [inline src/eFAST.jl]: what's this parameter?

File: src/eFAST.jl [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    s = collect((2*pi/n) * (0:n-1))
+    ps = zeros(num_params,n*num_params)
+
+    for i in 1:num_params
+        omega_temp[i] = omega[1]
+        omega_temp[[k for k in 1:num_params if k != i]] = omega[2:end]
+        l = collect((i-1)*n+1:i*n)
+        phi = 2*pi*rand()
+        for j in 1:num_params
+            x =  0.5 .+ (1/pi) .*(asin.(sin.(omega_temp[j]*s .+ phi)))
```
Comment (ChrisRackauckas) [inline src/eFAST.jl]: `asin(sin` is just to take the mod?

Comment (Vaibhavdixit02) [reply src/eFAST.jl]: Not always? 
```
julia> asin(sin(10))
-0.5752220392306202

julia> asin(sin(pi/2))
1.5707963267948966
```

Comment (ChrisRackauckas) [reply src/eFAST.jl]: looks like that's a mod 2pi? If not, what's it doing? It's a fairly odd piece of code.

File: src/eFAST.jl [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            ps[j,l] .= quantile.(Uniform(p_range[j][1], p_range[j][2]),x)
```
Comment (ChrisRackauckas) [inline src/eFAST.jl]: why quantile, and why broadcast?

File: src/eFAST.jl (hunk 1 above)
Comment (Vaibhavdixit02) [reply src/eFAST.jl]: The number of harmonics to sum in the Fourier series decomposition

File: src/eFAST.jl (hunk 2 above)
Comment (Vaibhavdixit02) [reply src/eFAST.jl]: Oh you mean modulus, sorry, I understood it as `abs`. Yes then it's the same thing.

File: src/eFAST.jl [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+struct eFASTResult{T1}
+    first_order::T1
+    total_order::T1
+end
+
+function gsa(f,method::eFAST,p_range::AbstractVector,n::Int=1000;batch=false)
+    @unpack num_harmonics = method
+    num_params = length(p_range)
+    omega = [floor((n-1)/(2*num_harmonics))]
```
Comment (ChrisRackauckas) [inline src/eFAST.jl]: where is this choice coming from?

**Pull request merged**